In [31]:
import pandas as pd

In [32]:
pd.set_option('display.max_columns', None) # 컬럼 생략 금지
pd.set_option('display.max_rows', 100) # 행 100개까지 표시
pd.options.display.float_format='{:.0f}'.format # float data 소수점 이하 날리고 표시

In [36]:
df = pd.read_csv("data/real_estate_total_edit.csv", encoding="euc-kr")
df.head()

C:\Users\gorhk\AppData\Local\Temp\ipykernel_5068\569592905.py:1: DtypeWarning: Columns (14) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("data/real_estate_total_edit.csv", encoding="euc-kr")


CGG_CD CGG_NM  STDG_CD STDG_NM        BLDG_NM    CTRT_DAY  THING_AMT  \
0   11500    강서구    10300     화곡동  우장산아이파크,이편한세상  2019-04-25      95000   
1   11410   서대문구    11200     대현동           럭키대현  2019-03-16      70000   
2   11590    동작구    10700     사당동             현대  2019-02-21      49500   
3   11350    노원구    10500     상계동           초산빌라  2019-01-21      16000   
4   11380    은평구    11000     증산동   궁전맨션(189-63)  2019-01-03      43000   

   ARCH_AREA  LAND_AREA  FLR RGHT_SE  RTRCN_DAY  ARCH_YR BLDG_USG DCLR_SE  
0        102        NaN   10     NaN        NaN     2008      아파트     NaN  
1         83        NaN   10     NaN        NaN     1999      아파트     NaN  
2         52        NaN   10     NaN        NaN     1991      아파트     NaN  
3         41         18    1     NaN        NaN     1996    연립다세대     NaN  
4         54         27    2     NaN        NaN     1992    연립다세대     NaN

- 접수연도의 경우 부동산 가격 분석에 있어서 크게 의미있는 변수는 아니라고 생각하여 버려도 될 것 같습니다! -> 계약일이 중요
- 계약일 - 건축연도 = 계약 당시 건물 연식

- 건물용도 4종류(아파트, 연립다세대, 오피스텔, 단독다가구) 확인

In [37]:
print(df['BLDG_USG'].unique())
print(df.info())

['아파트' '연립다세대' '오피스텔' '단독다가구']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 833614 entries, 0 to 833613
Data columns (total 15 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   CGG_CD     833614 non-null  int64  
 1   CGG_NM     833613 non-null  object 
 2   STDG_CD    833614 non-null  int64  
 3   STDG_NM    833614 non-null  object 
 4   BLDG_NM    776065 non-null  object 
 5   CTRT_DAY   833614 non-null  object 
 6   THING_AMT  833614 non-null  int64  
 7   ARCH_AREA  833614 non-null  float64
 8   LAND_AREA  668824 non-null  float64
 9   FLR        776119 non-null  float64
 10  RGHT_SE    7614 non-null    object 
 11  RTRCN_DAY  25541 non-null   float64
 12  ARCH_YR    830198 non-null  float64
 13  BLDG_USG   833614 non-null  object 
 14  DCLR_SE    225344 non-null  object 
dtypes: float64(5), int64(3), object(7)
memory usage: 95.4+ MB
None


In [38]:
df2 = df.copy()

- "자치구명" 결측치 확인

In [40]:
df2.loc[df2["CGG_NM"].isna()]

CGG_CD CGG_NM  STDG_CD STDG_NM               BLDG_NM    CTRT_DAY  \
829987   26230    NaN    10100     미아동  2020년 2차 행복주택 입주자 모집  2024-01-12   

        THING_AMT  ARCH_AREA  LAND_AREA  FLR RGHT_SE  RTRCN_DAY  ARCH_YR  \
829987      42450         72          0    5     분양권        NaN        0   

       BLDG_USG DCLR_SE  
829987      아파트    중개거래

- "자치구명"의 결측치는 단 1건이고, 부산 소재 공인중개사의 단순 오기로 판단되어 dropna하는 것이 효율적이라 판단함.

In [43]:
# 해당 오류 행 제거
df2 = df2.dropna(subset=['CGG_NM'])

In [44]:
print("=== 데이터 정보 (Info) ===")
df2.info() # 컬럼별 데이터 개수(결측치 확인)와 데이터 타입(int, float, object)을 보여줍니다.

# 4. 데이터의 실제 생김새 확인하기 (상위 5줄)
print("\n=== 데이터 미리보기 (Head) ===")
display(df2.head()) # 데이터의 상위 5개 행을 출력하여 값의 형태를 눈으로 확인합니다.

# 5. 기초 통계량 확인하기 (숫자형 데이터)
print("\n=== 기초 통계량 (Describe) ===")
display(df2.describe()) # 평균, 표준편차, 최소/최대값 등을 한눈에 봅니다.

=== 데이터 정보 (Info) ===
<class 'pandas.core.frame.DataFrame'>
Index: 833613 entries, 0 to 833613
Data columns (total 15 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   CGG_CD     833613 non-null  int64  
 1   CGG_NM     833613 non-null  object 
 2   STDG_CD    833613 non-null  int64  
 3   STDG_NM    833613 non-null  object 
 4   BLDG_NM    776064 non-null  object 
 5   CTRT_DAY   833613 non-null  object 
 6   THING_AMT  833613 non-null  int64  
 7   ARCH_AREA  833613 non-null  float64
 8   LAND_AREA  668823 non-null  float64
 9   FLR        776118 non-null  float64
 10  RGHT_SE    7613 non-null    object 
 11  RTRCN_DAY  25541 non-null   float64
 12  ARCH_YR    830197 non-null  float64
 13  BLDG_USG   833613 non-null  object 
 14  DCLR_SE    225343 non-null  object 
dtypes: float64(5), int64(3), object(7)
memory usage: 101.8+ MB

=== 데이터 미리보기 (Head) ===


CGG_CD CGG_NM  STDG_CD STDG_NM        BLDG_NM    CTRT_DAY  THING_AMT  \
0   11500    강서구    10300     화곡동  우장산아이파크,이편한세상  2019-04-25      95000   
1   11410   서대문구    11200     대현동           럭키대현  2019-03-16      70000   
2   11590    동작구    10700     사당동             현대  2019-02-21      49500   
3   11350    노원구    10500     상계동           초산빌라  2019-01-21      16000   
4   11380    은평구    11000     증산동   궁전맨션(189-63)  2019-01-03      43000   

   ARCH_AREA  LAND_AREA  FLR RGHT_SE  RTRCN_DAY  ARCH_YR BLDG_USG DCLR_SE  
0        102        NaN   10     NaN        NaN     2008      아파트     NaN  
1         83        NaN   10     NaN        NaN     1999      아파트     NaN  
2         52        NaN   10     NaN        NaN     1991      아파트     NaN  
3         41         18    1     NaN        NaN     1996    연립다세대     NaN  
4         54         27    2     NaN        NaN     1992    연립다세대     NaN


=== 기초 통계량 (Describe) ===


CGG_CD  STDG_CD  THING_AMT  ARCH_AREA  LAND_AREA    FLR  RTRCN_DAY  \
count  833613   833613     833613     833613     668823 776118      25541   
mean    11451    10941      64248         70         32      7   20217012   
std       173     1252      75159         66        394      6      13895   
min     11110    10100       1700          5          0     -3   20200221   
25%     11305    10200      24500         39          0      3   20201129   
50%     11470    10500      42000         59         23      5   20211116   
75%     11590    10900      80000         85         35     10   20230502   
max     11740    18700   11087780       3620     294294     73   20241014   

       ARCH_YR  
count   830197  
mean      1984  
std        191  
min          0  
25%       1993  
50%       2003  
75%       2013  
max       2024

층의 min값이 -3 음수인 것을 확인. 이상치 확인 필요.

In [21]:
df_total2.loc[df_total2["층"]==-3]

접수연도  자치구코드 자치구명  법정동코드 법정동명  지번구분 지번구분명  본번  부번  \
56832   2018  11380  은평구  11400  진관동     1    대지  84   0   
165577  2018  11380  은평구  11400  진관동     1    대지  84   0   
359024  2020  11380  은평구  11400  진관동     1    대지  84   0   
430119  2020  11380  은평구  11400  진관동     1    대지  84   0   
713893  2023  11380  은평구  11400  진관동     1    대지  84   0   
750174  2023  11170  용산구  13100  한남동     1    대지  15  12   

                              건물명        계약일    물건금액  건물면적  토지면적  층 권리구분  취소일  \
56832   우물골2단지두산위브(223~244동)BL2-7 2018-07-26   72500   101     0 -3  NaN  NaN   
165577  우물골2단지두산위브(223~244동)BL2-7 2017-11-16   70500   135     0 -3  NaN  NaN   
359024  우물골2단지두산위브(223~244동)BL2-7 2020-07-23  104000   135     0 -3  NaN  NaN   
430119         은평뉴타운 우물골(두산위브7단지) 2020-03-20   82000   135   NaN -3  NaN  NaN   
713893  우물골2단지두산위브(223~244동)BL2-7 2023-07-29  110000   101     0 -3  NaN  NaN   
750174                      코번하우스 2023-02-09  580000   274   211 -3  NaN  NaN   

        건축년도   건물용도  신고구분   중개사위치  
56832   2010    아파트   NaN     NaN  
165577  2010    아파트   NaN     NaN  
359024  2010    아파트   NaN     NaN  
430119  2010    아파트   NaN     NaN  
713893  2010    아파트  중개거래  서울 은평구  
750174  2006  연립다세대  중개거래  서울 용산구

- -1~-2층은 반지하 매물 가능성 있음 -> 이상치 제외 X
- -3층: 특정 매물(건물명: 우물골2단지두산위브(223~244동)BL2-7 / 은평구 진관동)의 경우 산을 깎아 만든 비탈 지형 매물로 실 거주 환경은 지상층과 같거나 오히려 넓은 테라스가 딸린 로열층임메도 서류상 표기만 -3층일 가능성 존재. 2018.07 매매가 평균 6.35억임에 반해 물건 금액이 7억이 넘는 것이 이를 뒷받침. -> 이상치 제외 X // 다만, 반지하 구분 시 해당 매물은 반지하 매물과 분리할 필요 있음.
- 추가 조사 결과, 은평구 진관동 은평뉴타운, 용산구 한남동 코번하우스의 경우도 동일하게 고급 주거 형태임을 확인함.
- 동일한 논리로 -1, -2층의 경우도 조사. 조사 결과, 실제 반지하와 고급 주거 형태 혼재. '물건가격' 컬럼 등을 활용해 둘을 분리할 필요 있음.
- 결론: 가격대를 기준으로 지하층의 매물 반지하/고급주거형태 구분 필요함.

- '지번구분' isna & '건물명' isna => 57000건에 달함. 이것을 drop하는게 맞을지 fill하는게 맞을지 모르겠음.

In [ ]:
# 얘들은 매물 주소 검색이 아예 안될 애들이라 어쩔 수 없이 전부 drop해야될 것 같습니다.
df_total2.loc[(df_total2['지번구분'].isna())&(df_total2['본번'].isna())&(df_total2['건물명'].isna())]

접수연도  자치구코드  자치구명  법정동코드    법정동명  지번구분 지번구분명  본번  부번  건물명        계약일  \
14      2018  11530   구로구  10800     오류동   NaN   NaN NaN NaN  NaN 2018-12-31   
16      2018  11215   광진구  10100     중곡동   NaN   NaN NaN NaN  NaN 2018-12-31   
43      2018  11560  영등포구  10900  영등포동8가   NaN   NaN NaN NaN  NaN 2018-12-29   
87      2018  11230  동대문구  10900     휘경동   NaN   NaN NaN NaN  NaN 2018-12-28   
92      2018  11620   관악구  10200     신림동   NaN   NaN NaN NaN  NaN 2018-12-27   
...      ...    ...   ...    ...     ...   ...   ...  ..  ..  ...        ...   
833379  2024  11620   관악구  10200     신림동   NaN   NaN NaN NaN  NaN 2023-12-08   
833396  2024  11680   강남구  11100     세곡동   NaN   NaN NaN NaN  NaN 2023-12-07   
833579  2024  11230  동대문구  10500    답십리동   NaN   NaN NaN NaN  NaN 2023-12-02   
833612  2024  11440   마포구  11000    노고산동   NaN   NaN NaN NaN  NaN 2021-07-19   
833613  2024  11305   강북구  10100     미아동   NaN   NaN NaN NaN  NaN 2021-02-08   

          물건금액  건물면적  토지면적   층 권리구분      취소일  건축년도   건물용도  신고구분    중개사위치  
14       73000   197   146 NaN  NaN      NaN  1990  단독다가구   NaN      NaN  
16      102000    89   131 NaN  NaN      NaN  1973  단독다가구   NaN      NaN  
43       88700    36    92 NaN  NaN      NaN  1945  단독다가구   NaN      NaN  
87       95000   140   112 NaN  NaN      NaN  1991  단독다가구   NaN      NaN  
92       56500   179   127 NaN  NaN      NaN  1988  단독다가구   NaN      NaN  
...        ...   ...   ...  ..  ...      ...   ...    ...   ...      ...  
833379   75000   141   132 NaN  NaN      NaN  1985  단독다가구  중개거래   서울 관악구  
833396  520000   166   461 NaN  NaN 20240117  1982  단독다가구  중개거래   서울 송파구  
833579   23000    15    34 NaN  NaN      NaN  1900  단독다가구  중개거래  서울 동대문구  
833612  195000   157   117 NaN  NaN 20240417  1988  단독다가구   NaN      NaN  
833613  164000   293   136 NaN  NaN 20240905  1994  단독다가구   NaN      NaN  

[57482 rows x 21 columns]

In [23]:
# 지번구분이 na, 본번 na, 건물명 살아있는 데이터
# 건물명이 있어서 검색이 가능하며 살릴 수 있다!!
# 심지어 건물 두개만 이러함(힐스테이트 서초 젠트리스, 자연누리 오피스텔) / 두 건물 법정동도 서초구 신원동으로 동일
target_data=df_total2.loc[(df_total2['지번구분'].isna())&(df_total2['본번'].isna())&df_total2['건물명'].notna()]
print(target_data.shape)
print(target_data["건물명"].nunique())
print(target_data["건물명"].unique())

(76, 21)
2
['힐스테이트 서초 젠트리스' '자연누리 오피스텔']


In [24]:
# 

target_data2 = df_total2.loc[(df_total2['지번구분'].isna())&(df_total2['본번'].notna())]
print(target_data2.shape)
display(target_data2["건물명"].unique())
display(target_data2["본번"].unique())

(13, 21)


array(['(239-40)', 'KSK빌1차', '늘사랑교회', '(34-3)', '(102-23)',
       'DiorHouse664', '보성Family', '597-68', '동서빌라(510-19)', '(11-11)',
       '(807-8)', '진흥푸르지오', '(794-16)'], dtype=object)

array([ 239.,  211., 1068.,   34.,  102.,  664.,  909.,  597.,  510.,
         11.,  807.,   26.,  794.])

In [25]:
# 문제의 57,482개 데이터만 따로 추출 (지번구분도 없고, 건물명도 없는 애들)
df_problem = df_total2[
    (df_total2['지번구분'].isnull()) & 
    (df_total2['건물명'].isnull())
]

print(df_problem['건물용도'].value_counts())

print("\n=== 본번(번지) 존재 여부 ===")
print(df_problem['본번'].notnull().value_counts())

건물용도
단독다가구    57482
Name: count, dtype: int64

=== 본번(번지) 존재 여부 ===
본번
False    57482
Name: count, dtype: int64


In [26]:
target_data3 = df_total2.loc[(df_total2['법정동코드'].isna())|(df_total2['자치구명'].isna())]
print(target_data3.shape)
display(target_data3["건물명"].unique())
display(target_data3["본번"].unique())

(0, 21)


array([], dtype=object)

array([], dtype=float64)

In [27]:
# 신고구분의 종류는?

counts = df_total['신고구분'].value_counts()
counts

신고구분
중개거래    184042
직거래      41302
Name: count, dtype: int64